[Bug] Fix some definition errors

VisualSJ opened the pull request:
Re: #

### Changelog

* Fix some definition errors

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

<!-- Note: Makes sure these boxes are checked before submitting your PR - thank you!
- [ ] Your pull request title is using English, it's precise and appropriate.
- [ ] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [ ] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- [ ] Document new code with comments in source code based on API docs
- [ ] Make sure any runtime log information in `log` , `error` or `new Error('')` has been moved into `EngineErrorMap.md` with an ID, and use `logID(id)` or `new Error(getError(id))` instead.
- To official teams:
  - [ ] Check that your PR is following our [guides](https://github.com/cocos/3d-tasks/blob/master/workflows/readme.md)
-->

VisualSJ committed:
[Bug] Fix some definition errors

# Conflicts:
#	cocos/spine/lib/spine-core.d.ts
#	package.json

dumganhar commented on cocos/core/utils/array.ts:
```diff
@@ -24,7 +24,71 @@
+
+    public push (item: T): void {
+        this.array.push(item);
+    }
```
Why move the file content of `mutable-forward-iterator.ts` to here?

VisualSJ replied:
When compiling the dts, this definition will be lost, so it will be normal to put it here.

dumganhar commented on cocos/gfx/base/pipeline-state.editor.ts:
```diff
@@ -319,7 +319,7 @@ export class PassStatesEditor implements EffectAsset.IPassStates {
     @serializable
     @editable
-    public blendState: BlendStateEditor = new BlendStateEditor();
+    public blendState: any = new BlendStateEditor();
```
Why change to `any`?

VisualSJ replied:
The definition here is incorrect, and I don't know how to solve it. The base type is expecting a BlendState, but a BlendStateEditor is given here. In fact, there have been errors in this code for a long time.

dumganhar commented on cocos/spine/lib/spine-core.d.ts:
```diff
@@ -307,7 +307,7 @@ declare namespace spine {
     class AnimationStateData {
         skeletonData: SkeletonData;
-        animationToMixTime: Map<number>;
+        animationToMixTime: Map<string, number>;
```
@bofeng-song Plz review the spine changes.

bofeng-song replied:
different with https://github.com/EsotericSoftware/spine-runtimes/blob/3.8/spine-ts/build/spine-core.d.ts

VisualSJ replied:
The definition here is not wrong, but when cocos outputs dts, it does not add the namespace, which causes the map of spine to pollute other code.

dumganhar commented on cocos/spine/lib/spine-core.d.ts:
```diff
@@ -760,7 +760,7 @@ declare namespace spine {
+        MipMap = 9930,
         MipMapNearestNearest = 9984,
         MipMapLinearNearest = 9985,
         MipMapNearestLinear = 9986,
```
https://github.com/EsotericSoftware/spine-runtimes/blob/b09e189c6438cb9358005771c3304b8f52bc4e30/spine-ts/spine-core/src/Texture.ts#L49C5-L49C5
```ts
export enum TextureFilter {
	Nearest = 9728, // WebGLRenderingContext.NEAREST
	Linear = 9729, // WebGLRenderingContext.LINEAR
	MipMap = 9987, // WebGLRenderingContext.LINEAR_MIPMAP_LINEAR
	MipMapNearestNearest = 9984, // WebGLRenderingContext.NEAREST_MIPMAP_NEAREST
	MipMapLinearNearest = 9985, // WebGLRenderingContext.LINEAR_MIPMAP_NEAREST
	MipMapNearestLinear = 9986, // WebGLRenderingContext.NEAREST_MIPMAP_LINEAR
	MipMapLinearLinear = 9987 // WebGLRenderingContext.LINEAR_MIPMAP_LINEAR
}
```

Why change the MipMap value, it's wrong. The configuration of eslint which ensure the enum value should be different is too strict.

VisualSJ replied:
The values of MipMap and MipMapLinearLinear are the same. I have only changed the wrong parts to other values. If you know the correct value, you should change it to the value you know.

bofeng-song commented on cocos/spine/lib/spine-core.d.ts:
```diff
@@ -338,7 +338,7 @@ declare namespace spine {
         dispose(): void;
         hasErrors(): boolean;
-        getErrors(): Map<string>;
+        getErrors(): Map<string, string>;
```
different with https://github.com/EsotericSoftware/spine-runtimes/blob/3.8/spine-ts/build/spine-core.d.ts

VisualSJ replied:
The definition here is not wrong, but when cocos outputs dts, it does not add the namespace, which causes the map of spine to pollute other code.

bofeng-song commented on cocos/spine/lib/spine-core.d.ts:
```diff
@@ -544,7 +544,7 @@ declare namespace spine {
         dispose(): void;
         hasErrors(): boolean;
-        getErrors(): Map<string>;
+        getErrors(): Map<string, string>;
```
different with https://github.com/EsotericSoftware/spine-runtimes/blob/3.8/spine-ts/build/spine-core.d.ts

VisualSJ replied:
The definition here is not wrong, but when cocos outputs dts, it does not add the namespace, which causes the map of spine to pollute other code.

bofeng-song commented on cocos/spine/lib/spine-core.d.ts:
```diff
@@ -705,7 +705,7 @@ declare namespace spine {
     class Skin {
         name: string;
-        attachments: Map<Attachment>[];
+        attachments: Map<string, Attachment>[];
```
different with https://github.com/EsotericSoftware/spine-runtimes/blob/3.8/spine-ts/build/spine-core.d.ts

VisualSJ replied:
The definition here is not wrong, but when cocos outputs dts, it does not add the namespace, which causes the map of spine to pollute other code.

qiuguohua committed:
Fix the error when exiting audio on Xiaomi platform. (#16577)

Co-authored-by: qiuguohua <guohua.qiu@cocos.com>

star-e committed:
V3.8.2 pipeline simplify (#16578)

knoxHuang committed:
Fix label outline error causing ineffective outline. (#16583)

star-e committed:
V3.8.2 comment (#16581)

GengineJS committed:
Modify uniform offset (#16588)

yanOO1497 committed:
fix compressZip in invaild in native platform (#16596)

bofeng-song committed:
Fix errors caused by reusing previously freed objects(skeletonInfo.skeleton) (#16605)

bofeng-song committed:
Fix setAnimation can not work normally at cache mode while animation's name different with default. (#16603)

GengineJS committed:
Fix instanced buffer memory leak problem (#16609)

dream93 committed:
Update MiddlewareManager.cpp (#16597)

fix spine crash

dumganhar committed:
Fix cc.assetManager.downloader.maxConcurrency is overrode by mistake on mini-game platforms. (#16565)

bofeng-song committed:
Refine:  make multi skeleton component shared one instance of spine.Skeleton while spine working at sharedCache mode (#16593)

* Refine:  make multi skeleton component shared one instance of spine.Skeleton while spine working at sharedCache mode

Y-way committed:
Fix building error for simulator. (#16612)

* Update .cocos-project.json

* Fix build error

knoxHuang committed:
Fix baking data being emptied when switching scenes (#16613)

Fixed that when switching scenes, re-adding the light probe group component will empty the baking data.

qiuguohua committed:
Fixed an issue where fntConfig may be null and report an error. (#16618)

Co-authored-by: qiuguohua <guohua.qiu@cocos.com>

knoxHuang committed:
Fix scene dirty after ReflectionProbe Bake. (#16619)

knoxHuang committed:
fix unregister child events when removing 'light-probe-group-footer' (#16617)

* fix unregister child events when removing 'light-probe-group-footer'

* refine code

dumganhar committed:
Update web templates, check wether polyfill file exists in index.html. (#16614)

hana-alice committed:
missing shader visibility for CCLocal (#16506)

knoxHuang committed:
fix coefficients array not init when "updateProbes" needs to add a new probe (#16627)

finscn committed:
improve `cc.Color.fromHEX()` (#16561)

* improve `cc.Color.fromHEX()`

* support hex-number
* better performance

finscn committed:
Use `clearTracks()` instead of `clearTrack(0)` (#16606)

qiuguohua committed:
Update the position of the input if the size changes. (#16634)

Co-authored-by: qiuguohua <guohua.qiu@cocos.com>

knoxHuang committed:
upgraded default asset scene version to 1.1.50 (#16636)

knoxHuang committed:
update label outline and shadow deprecated log (#16637)

* update label outline and shadow deprecated log

bofeng-song committed:
Fix the issue of incorrect updating of the Spine shared-cache reference count. (#16639)

star-e committed:
improve performance (#16650)

* improve performance

* fix typo

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: e31656893aca28cf32daa0d8277d7596b5a63bbd  (#16650) (#16655)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

bofeng-song committed:
Fixing the reference counting bug in Spine. (#16663)

star-e committed:
V3.8.2 gfx (#16664)

* fix glsl macro bug

* lint

VisualSJ committed:
review code
